flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/formats/raw/RawFormatDeserializationSchema.java

```diff
@@ -0,0 +1,343 @@
+ */
+@Internal
+public class RawFormatDeserializationSchema implements DeserializationSchema<RowData> {
+
+	private static final long serialVersionUID = 1L;
+
+	private final LogicalType deserializedType;
+
+	private final DeserializationRuntimeConverter converter;
+
+	private final TypeInformation<RowData> producedTypeInfo;
+
+	private final String charsetName;
+
+	private final boolean isBigEndian;
+
+	private transient GenericRowData reuse;
+
+	public RawFormatDeserializationSchema(
+			LogicalType deserializedType,
+			TypeInformation<RowData> producedTypeInfo,
+			String charsetName,
+			boolean isBigEndian) {
+		this.deserializedType = checkNotNull(deserializedType);
+		this.producedTypeInfo = checkNotNull(producedTypeInfo);
+		this.converter = createConverter(deserializedType, charsetName, isBigEndian);
+		this.charsetName = charsetName;
+		this.isBigEndian = isBigEndian;
+	}
+
+	@Override
+	public void open(InitializationContext context) throws Exception {
+		reuse = new GenericRowData(1);
+		converter.open();
+	}
+
+	@Override
+	public RowData deserialize(byte[] message) throws IOException {
+		reuse.setField(0, converter.convert(message));
+		return reuse;
+	}
+
+	@Override
+	public boolean isEndOfStream(RowData nextElement) {
+		return false;
+	}
+
+	@Override
+	public TypeInformation<RowData> getProducedType() {
+		return producedTypeInfo;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		RawFormatDeserializationSchema that = (RawFormatDeserializationSchema) o;
+		return producedTypeInfo.equals(that.producedTypeInfo) &&
+			deserializedType.equals(that.deserializedType) &&
+			charsetName.equals(that.charsetName) &&
+			isBigEndian == that.isBigEndian;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(producedTypeInfo, deserializedType, charsetName, isBigEndian);
+	}
+
+	// ------------------------------------------------------------------------
+
+	/**
+	 * Runtime converter that convert byte[] to internal data structure object.
+	 */
+	@FunctionalInterface
+	private interface DeserializationRuntimeConverter extends Serializable {
+
+		default void open() {}
+
+		Object convert(byte[] data) throws IOException;
+	}
+
+	/**
+	 * Creates a runtime converter.
+	 */
+	private static DeserializationRuntimeConverter createConverter(
+			LogicalType type, String charsetName, boolean isBigEndian) {
+		final DeserializationRuntimeConverter converter = createNotNullConverter(type, charsetName, isBigEndian);
+		final Consumer<byte[]> validator = createDataLengthValidator(type);
+
+		return new DeserializationRuntimeConverter() {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			public void open() {
+				converter.open();
+			}
+
+			@Override
+			public Object convert(byte[] data) throws IOException {
+				if (data == null) {
```

Comment:
I think data is never null at this location.

Reply:
Why? The Kafka message value might be null.

Reply:
at other locations we also don't check that:
`JsonRowDataDeserializationSchema#deserialize` or `CsvRowDataDeserializationSchema#deserialize`

the JavaDocs of `DeserializationSchema` don't mention that either

Reply:
but let's keep the null check for now

Reply:
I think both JSON and CSV format didn't implement correctly. There will be NPE if encountering tombstone messages. That can be skipped when `ignoreParseErrors` is enabled, but I think the format should handle tombstone messages.